snuba/subscriptions/entity_subscription.py

```diff
@@ -0,0 +1,122 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, List, Mapping, Optional, Sequence, Type
+
+from snuba.datasets.entities import EntityKey
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.exceptions import InvalidQueryException
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.types import Condition
+
+
+class SubscriptionType(Enum):
+    LEGACY = "legacy"
+    SNQL = "snql"
+    DELEGATE = "delegate"
+
+
+class EntitySubscription(ABC):
+    def __init__(self, data_dict: Mapping[str, Any]) -> None:
+        ...
+
+    @abstractmethod
+    def get_entity_subscription_conditions_for_snql(
+        self, offset: Optional[int] = None
+    ) -> List[Expression]:
+        """
+        Returns a list of extra conditions that are entity specific and required for the
+        snql subscriptions
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_entity_subscription_conditions_for_legacy(
+        self, offset: Optional[int] = None
+    ) -> Sequence[Condition]:
+        """
+        Returns a list of extra conditions that are entity specific and required for the
+        legacy subscriptions
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def to_dict(self) -> Mapping[str, Any]:
+        raise NotImplementedError
+
+
+class SessionsSubscription(EntitySubscription):
+    def __init__(self, data_dict: Mapping[str, Any]) -> None:
+        super().__init__(data_dict)
```

Comment:
No need for this

Reply:
the linter complains if I don't define this

Reply:
This needs to be applied together with this other comment https://github.com/getsentry/snuba/pull/2094#discussion_r715083833

Reply:
I cannot remove it because of this https://github.com/getsentry/snuba/pull/2094#discussion_r715089069

Reply:
Ah, forgot you're always passing the dict to the constructor